doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,834 @@
+
+.. figure:: conv_arithmetic_figures/no_padding_strides_transposed.gif
+
+This should help understand what fractional strides involve: zeros
+are inserted *between* input units, which makes the kernel move around at
+a slower pace than with unit strides.
+
+.. note::
+
+    Doing so is inefficient and real-world implementations avoid useless
+    multiplications by zero, but conceptually it is how the transpose of a
+    strided convolution can be thought of.
+
+For the moment, it will be assumed that the convolution is non-padded (:math:`p
+= 0`) and that its input size :math:`i` is such that :math:`i - k` is a multiple
+of :math:`s`. In that case, the following relationship holds:
+
+.. admonition:: Relationship 11
+
+    A convolution described by :math:`p = 0`, :math:`k` and :math:`s` and whose
+    input size is such that :math:`i - k` is a multiple of :math:`s`, has an
+    associated transposed convolution described by :math:`\tilde{i}'`, :math:`k'
+    = k`, :math:`s' = 1` and :math:`p' = k - 1`, where :math:`\tilde{i}'` is the
+    size of the stretched input obtained by adding :math:`s - 1` zeros between
+    each input unit, and its output size is
+
+    .. math::
+
+        o' = s (i' - 1) + k.
+
+    In other words,
+
+    .. code-block:: python
+
+        input = theano.tensor.nnet.conv2d_grad_wrt_inputs(
+            output, filters, filter_shape=(c1, c2, k, k), border_mode=(0, 0),
+            subsample=(s, s))
+        # input.shape[2] == input.shape[3] == s * (output.shape[2] - 1) + k
+
+Zero padding, non-unit strides, transposed
+------------------------------------------
+
+When the convolution's input size :math:`i` is such that :math:`i + 2p - k` is a
+multiple of :math:`s`, the analysis can extended to the zero padded case by
+combining Relationship 8 and Relationship 11:
+
+.. admonition:: Relationship 12
+
+    A convolution described by :math:`k`, :math:`s` and :math:`p` and whose
+    input size :math:`i` is such that :math:`i + 2p - k` is a multiple of
+    :math:`s` has an associated transposed convolution described by
+    :math:`\tilde{i}'`, :math:`k' = k`, :math:`s' = 1` and :math:`p' = k - p -
+    1`, where :math:`\tilde{i}'` is the size of the stretched input obtained by
+    adding :math:`s - 1` zeros between each input unit, and its output size is
+
+    .. math::
+
+        o' = s (i' - 1) + k - 2p.
+
+    In other words,
+
+    .. code-block:: python
+
+        o_prime = s * (output.shape[2] - 1) + k - 2 * p
+        input = theano.tensor.nnet.conv2d_grad_wrt_inputs(
+            output, filters, input_shape=(o_prime, o_prime),
+            filter_shape=(c1, c2, k, k), border_mode=(p, p),
+            subsample=(s, s))
+
+Here is an example for :math:`i = 5`, :math:`k = 3`, :math:`s = 2` and :math:`p
+= 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides_transposed.gif
+
+The constraint on the size of the input :math:`i` can be relaxed by introducing
```

Comment:
I think we should remind (probably at the beginning of the section) that when `s > 1` multiple direct convolutions can give the same output shape, due to the `floor` operator in the formula.